### PR TITLE
Implementation of transformation function and fixed test resume functionality

### PR DESCRIPTION
**Pull request by Deepankar-promact**
**Fixed Issues**

#22936

**Files Changed**

TestConductController.cs - Removed JSON serialisation.
ITestConductRepository.cs - Updated AddAnswerAsync function parameter.
TestConductRepository.cs - Fixed function for adding answers. Added Transformation function. 
test.component.ts - Fixed test resuming functionality.
test-end.component.ts - Added window close function on click of the button.
conduct.service.ts - Updated addAnswer function parameter.
test-end.html - Added angular binding.

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

**Known Issues**

- No issue

**Comments**

NA

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/377)

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 1]
```diff
@@ -160,5 +204,64 @@ public TestConductRepository(TrappistDbContext dbContext)
...
+                    {
+                        var testAnswers = new TestAnswers();
+                        //Adding attempted Question to TestConduct table
+                        var testConduct = new DomainModel.Models.TestConduct.TestConduct()
+                        {
+                            QuestionId = answer.QuestionId,
+                            QuestionStatus = answer.QuestionStatus,
+                            TestAttendeeId = attendeeId
+                        };
+                        await _dbContext.TestConduct.AddAsync(testConduct);
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: MAJOR Merge this if statement with the enclosing one. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1066)

**Comment by chintans** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [TestConductRepository.cs#L143](https://github.com/Promact/trappist/blob/cd09415ad9bd04fd2f45565aee6fa97418db4434/Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs#L143): Use the opposite operator ('<=') instead. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1940)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 2]
```diff
@@ -91,23 +91,61 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
...
 
+        public async Task<bool> IsAnswerValidAsync(int attendeeId, TestAnswerAC answer)
+        {
+            var attendeeAnswer = await _dbContext.AttendeeAnswers.FindAsync(attendeeId);
+            if(attendeeAnswer.Answers != null)
+            {
+                var deserializedAnswer = JsonConvert.DeserializeObject<TestAnswerAC[]>(attendeeAnswer.Answers);
+                if(deserializedAnswer[0] != null)
+                {
+                    return !(deserializedAnswer.Where(x => x.QuestionId == answer.QuestionId && x.QuestionStatus == QuestionStatus.answered).Count() > 0);
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: MINOR Use the opposite operator ('<=') instead. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1940)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Trappist/src/Promact.Trappist.Test/TestConduct/TestConductRepositoryTest.cs [hunk 3]
```diff
@@ -303,6 +304,55 @@ public TestConductRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
...
+            categoryListAc[0].IsSelect = true;
+            await _testRepository.AddTestCategoriesAsync(test.Id, categoryListAc);
+            //Creating test questions
+            var questionList = new List<QuestionAC>
+            {
+                CreateQuestionAC(true, "Category1 type question 1", category1.Id, 1),
+                CreateQuestionAC(true, "Category1 type question 2", category1.Id, 1),
+            };
+            await _testRepository.AddTestQuestionsAsync(questionList, test.Id);
+            var testInstruction = await _testConductRepository.GetTestInstructionsAsync(_stringConstants.MagicString);
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Test/TestConduct/TestConductRepositoryTest.cs]: MAJOR Remove this unused 'testInstruction' local variable. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)

**Comment by chintans** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 4]
```diff
@@ -91,23 +87,61 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
...
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: MINOR Use the opposite operator ('<=') instead. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1940)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 5]
```diff
@@ -172,5 +213,64 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
             return report.TestStatus;
         }
         #endregion
+
+        #region Private Method
+
+        private async Task TransformAttendeeAnswer(int attendeeId)
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: Add proper comments

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 6]
```diff
@@ -89,23 +86,61 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
...
 
+        public async Task<bool> IsAnswerValidAsync(int attendeeId, TestAnswerAC answer)
+        {
+            var attendeeAnswer = await _dbContext.AttendeeAnswers.FindAsync(attendeeId);
+            if(attendeeAnswer.Answers != null)
+            {
+                var deserializedAnswer = JsonConvert.DeserializeObject<TestAnswerAC[]>(attendeeAnswer.Answers);
+                if(deserializedAnswer[0] != null)
+                {
+                    return !(deserializedAnswer.Where(x => x.QuestionId == answer.QuestionId && x.QuestionStatus == QuestionStatus.answered).Count() > 0);
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: MINOR Use the opposite operator ('<=') instead. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1940)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**